### DISCUSSION TITLE
create template fails after two successful builds

### DISCUSSION THREAD

**Discussion by bharattalekar** [PaloAltoNetworks/pan-os-ansible#450] [Q&A]
vars_files:
    - templates.yml

tasks:
 - name: Create template
    panos_template:
      provider: "{{ provider }}"
      name: '{{ item.name }}'
      description: '{{ item.description }}'
      state: present
    loop: '{{ templates }}'

**This is A VAR FILE**

templates:
 - name: "AUTH-BASE"
   description: "Contains the Authentication Template for User Based Access"

 - name: "name1"
   description: "Contains  device specific configuration"

 - name: "name3"
   description: "Contains device specific configuration"

 - name: "name4"
   description: "This template will include the base device and networking configuration "

 - name: "name5"
   description: "Contains device specific configuration"

 - name: "name6"
   description: "Contains device specific configuration"
   
 - name: "name7"
   description: "Contains device specific configuration"

**error i get after two or three sucessful run** 

{
    "module_stdout": "",
    "module_stderr": "Traceback (most recent call last):\n  File \"/root/.ansible/tmp/ansible-tmp-1686854058.5282977-11397-270149301347371/AnsiballZ_panos_template.py\", line 102, in <module>\n    _ansiballz_main()\n  File \"/root/.ansible/tmp/ansible-tmp-1686854058.5282977-11397-270149301347371/AnsiballZ_panos_template.py\", line 94, in _ansiballz_main\n    invoke_module(zipped_mod, temp_path, ANSIBALLZ_PARAMS)\n  File \"/root/.ansible/tmp/ansible-tmp-1686854058.5282977-11397-270149301347371/AnsiballZ_panos_template.py\", line 40, in invoke_module\n    runpy.run_module(mod_name='ansible_collections.paloaltonetworks.panos.plugins.modules.panos_template', init_globals=None, run_name='__main__', alter_sys=True)\n  File \"/usr/lib64/python3.6/runpy.py\", line 205, in run_module\n    return _run_module_code(code, init_globals, run_name, mod_spec)\n  File \"/usr/lib64/python3.6/runpy.py\", line 96, in _run_module_code\n    mod_name, mod_spec, pkg_name, script_name)\n  File \"/usr/lib64/python3.6/runpy.py\", line 85, in _run_code\n    exec(code, run_globals)\n  File \"/tmp/ansible_panos_template_payload_3xi87fb6/ansible_panos_template_payload.zip/ansible_collections/paloaltonetworks/panos/plugins/modules/panos_template.py\", line 133, in <module>\n  File \"/tmp/ansible_panos_template_payload_3xi87fb6/ansible_panos_template_payload.zip/ansible_collections/paloaltonetworks/panos/plugins/modules/panos_template.py\", line 129, in main\n  File \"/tmp/ansible_panos_template_payload_3xi87fb6/ansible_panos_template_payload.zip/ansible_collections/paloaltonetworks/panos/plugins/module_utils/panos.py\", line 484, in process\n  File \"/tmp/ansible_panos_template_payload_3xi87fb6/ansible_panos_template_payload.zip/ansible_collections/paloaltonetworks/panos/plugins/module_utils/panos.py\", line 656, in apply_state\n  File \"/tmp/ansible_panos_template_payload_3xi87fb6/ansible_panos_template_payload.zip/ansible_collections/paloaltonetworks/panos/plugins/module_utils/panos.py\", line 79, in eltostr\n  File \"/opt/awx/envs/cloud-networking/lib/python3.6/site-packages/panos/base.py\", line 533, in element_str\n    parsed = minidom.parseString(raw)\n  File \"/usr/lib64/python3.6/xml/dom/minidom.py\", line 1968, in parseString\n    return expatbuilder.parseString(string)\n  File \"/usr/lib64/python3.6/xml/dom/expatbuilder.py\", line 925, in parseString\n    return builder.parseString(string)\n  File \"/usr/lib64/python3.6/xml/dom/expatbuilder.py\", line 223, in parseString\n    parser.Parse(string, True)\nxml.parsers.expat.ExpatError: not well-formed (invalid token): line 1, column 455\n",
    "exception": "Traceback (most recent call last):\n  File \"/root/.ansible/tmp/ansible-tmp-1686854058.5282977-11397-270149301347371/AnsiballZ_panos_template.py\", line 102, in <module>\n    _ansiballz_main()\n  File \"/root/.ansible/tmp/ansible-tmp-1686854058.5282977-11397-270149301347371/AnsiballZ_panos_template.py\", line 94, in _ansiballz_main\n    invoke_module(zipped_mod, temp_path, ANSIBALLZ_PARAMS)\n  File \"/root/.ansible/tmp/ansible-tmp-1686854058.5282977-11397-270149301347371/AnsiballZ_panos_template.py\", line 40, in invoke_module\n    runpy.run_module(mod_name='ansible_collections.paloaltonetworks.panos.plugins.modules.panos_template', init_globals=None, run_name='__main__', alter_sys=True)\n  File \"/usr/lib64/python3.6/runpy.py\", line 205, in run_module\n    return _run_module_code(code, init_globals, run_name, mod_spec)\n  File \"/usr/lib64/python3.6/runpy.py\", line 96, in _run_module_code\n    mod_name, mod_spec, pkg_name, script_name)\n  File \"/usr/lib64/python3.6/runpy.py\", line 85, in _run_code\n    exec(code, run_globals)\n  File \"/tmp/ansible_panos_template_payload_3xi87fb6/ansible_panos_template_payload.zip/ansible_collections/paloaltonetworks/panos/plugins/modules/panos_template.py\", line 133, in <module>\n  File \"/tmp/ansible_panos_template_payload_3xi87fb6/ansible_panos_template_payload.zip/ansible_collections/paloaltonetworks/panos/plugins/modules/panos_template.py\", line 129, in main\n  File \"/tmp/ansible_panos_template_payload_3xi87fb6/ansible_panos_template_payload.zip/ansible_collections/paloaltonetworks/panos/plugins/module_utils/panos.py\", line 484, in process\n  File \"/tmp/ansible_panos_template_payload_3xi87fb6/ansible_panos_template_payload.zip/ansible_collections/paloaltonetworks/panos/plugins/module_utils/panos.py\", line 656, in apply_state\n  File \"/tmp/ansible_panos_template_payload_3xi87fb6/ansible_panos_template_payload.zip/ansible_collections/paloaltonetworks/panos/plugins/module_utils/panos.py\", line 79, in eltostr\n  File \"/opt/awx/envs/cloud-networking/lib/python3.6/site-packages/panos/base.py\", line 533, in element_str\n    parsed = minidom.parseString(raw)\n  File \"/usr/lib64/python3.6/xml/dom/minidom.py\", line 1968, in parseString\n    return expatbuilder.parseString(string)\n  File \"/usr/lib64/python3.6/xml/dom/expatbuilder.py\", line 925, in parseString\n    return builder.parseString(string)\n  File \"/usr/lib64/python3.6/xml/dom/expatbuilder.py\", line 223, in parseString\n    parser.Parse(string, True)\nxml.parsers.expat.ExpatError: not well-formed (invalid token): line 1, column 455\n",
    "msg": "MODULE FAILURE\nSee stdout/stderr for the exact error",
    "rc": 1,
    "_ansible_no_log": false,
    "changed": false,
    "item": {
        "name": "name",
        "description": "Contains  device specific configuration"
    },
    "ansible_loop_var": "item",
    "_ansible_item_label": {
        "name": "name1",
        "description": "Contains device specific configuration"
    }
}

**Comment by jamesholland-uk** [comment]
Tracked in #451
